### DISCUSSION TITLE
WTF happened to Zed in last few updated?

### DISCUSSION THREAD

**Discussion by Frenkiee** [zed-industries/zed#21163] [Feature Requests]
What happened with Zed?! It went from the fastest IDE to the slowest one in two updates?! 

My project now has .ropeproject folders inside, and .mypy_cache was added without my approval...

The code suggestion is now open after half a minute?!

I'm seriously thinking of going back to VS code.

**Comment by jnptk** [comment]
Same here, what is `mccabe` and `pyflakes` all of a sudden doing in my project and creating folders without approval, I don't even use those!

Integration for more tools is cool and all, but they shouldn't be enabled by default IMO.

**Reply by jnptk** [reply]
> Integration for more tools is cool and all, but they shouldn't be enabled by default IMO.

Same for `pyright`, I wanna kill it and use ruff only.

**Reply by elisiariocouto** [reply]
I want to only use ruff as well, but at the moment it doesn't support navigation and autocompletion, which pyright and pylsp support. Latest Zed version tweaked some default settings, see https://github.com/zed-industries/zed/issues/20646

**Comment by osiewicz** [comment]
The new language server can be disabled with the following snippet in your settings.json:
```json
{
  "languages": {
    "Python": {
      "language_servers": ["!pylsp", "..."]
    }
 }
```
([source](https://github.com/zed-industries/zed/pull/20358#issuecomment-2493194836))
We've already fixed issues with `mypy_cache` being created; `ropeproject` one has to be fixed upstream. We would appreciate a repro for the slowness issues you're seeing.

**Reply by isatsam** [reply]
I'm observing a noticeable slowdown on a Ryzen 7 CPU/iGPU with a Python project that imports PySide6 and PIL libraries (among others, but these are the biggest two). It seems that with default settings, the new LSP just goes crazy analysing all imports and reports a ridiculous amount of issues in libraries:
![Screenshot_20241207_154453](https://github.com/user-attachments/assets/704241ba-8d02-4fd1-884e-c17ddff95bf3)
My own code fits within 1 file, so everything else it scans and reports is from the venv. I'm almost entirely sure these all have to be false positives. 
I'm also getting a lot of weird errors in my own code, when pyflakes would, for example, report a variable whose name I've not finished typing as non-existent, and then that error will just stick there for minutes. Same for mccabe, an error or a warning that's already fixed will still stay displayed for awhile.
Running `zed 0.162.5-1` on arch linux (if it's outdated, i'll update this evening and test again :)).
If you let me know what other information would be useful here, I'll open an issue about this. I could also upload the project (app.py + requirements.txt) if it would be more demonstrative.

**Comment by pchalasani** [comment]
My biggest frustrations with zed (mainly for Python) are these:

1. Basic code-completion for module imports is borked. E.g. when I do `from blah.` and pause at the "dot", I sometimes do not get a suggested list of modules, and I sometimes do, it's unpredictable
2. In the same example, when I do `from blah.foo import ` and pause after `import`, I expect good suggestions of modules to import (like in PyCharm), but I don't see them.
3. Random "project diagnostics" that I never asked for show up with an x-marked circle at the bottom. I had to hunt all installed extensions and uninstall them, and finally now it only has 2 errors from YAML files.
4. No docs on how to properly set up `pyright` (or should it be `pylsp`) and `ruff` correctly so that my code is not full of squiggles and also get proper auto-complete (1 and 2). I had to hunt thru discord, gh issues and reddit to find various `settings.json` snippets. Below is an extract from `settings.json`:

```json
 "languages": {
      "Python": {
        "language_servers": ["pyright", "ruff", "!python-refactoring", "!pylsp"],
        "enable_language_server": true,
        "format_on_save": "on",
        "formatter": [
          {
            "code_actions": {
              "source.organizeImports.ruff": true,
              "source.fixAll.ruff": true
            }
          },
          {
            "language_server": {
              "name": "ruff"
            }
          }
        ]
      }
    },
    
  "lsp": {
      "pyright": {
        "settings": {
          "python.analysis": {
            "ignore": ["*"],
            "diagnosticMode": "off",
            "typeCheckingMode": "off"
          },
          "python": {
            "pythonPath": ".venv/bin/python"
          }
        }
      },
      "ruff": {
            "initialization_options": {
              "settings": {
                // Ruff server settings goes here
                "lineLength": 89,
                "lint": {
                  "extendSelect": ["I"]
                }
              }
            }      
      },
      "pylsp": {
            "settings": {
              "plugins": {
                "rope": {
                  "enabled": true
                }
              }
            }
      },      
    },
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
